### DISCUSSION TITLE
¿Cómo puedo enviar un correo a los usuarios que pertenecen en una tasks sin valores en duro?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#117] [2. Questions and Answers] [answered]
Hola

Tengo que enviar un correo a los usuarios que participan en una tasks. Uno de esos usuarios es ingresado mediante por un formulario de asignación. Sé que con $Code# lo puedo hacer pero me faltaría un argumento

Lo tengo pensado de esta forma:

$CODE#task#channel#users#email

Gracias :)

**Comment by agallardol** [comment, marked as answer]
@NicolasPLG  Me hace sentido la forma en que está tratando de hacerlo. Intentaré ir explicándolo de forma incremental (iré complejizando poco a poco la expresión de cot-lang que nos permitirá obtener lo que necesitamos (lista de emails):

## Objetivo

Obtener lista con emails de los usuarios asociados a una tarea

## Pasos

- *Obtengamos la tarea que necesitamos*
  - **$CODE#task#channel#[channel-id]#[taskgroup-id]**
- *Obtengamos la lista de [user-id] de los participantes de la tarea en cuestión*
  - **($CODE#task#channel#[channel-id]#[taskgroup-id])|userList**
- *Con la lista de [user_id], obtengamos una lista de User*
  - **$CODE#users#_id#($CODE#task#channel#60648ebfa2c3680008b075c5#5e388e22c1346f6eb762f5d2)|userList**
- *Convirtamos la lista de User a una lista de User.email*
  - **($CODE#users#_id#($CODE#task#channel#60648ebfa2c3680008b075c5#5e388e22c1346f6eb762f5d2)|userList)|[map=>email]**
- *Enviemos el email que necesitamos a lista de emails*
  - *Completar campos en admin 🥇*

**Reply by agallardol** [reply]
Debes reemplazar los parámetros:

- [channel-id]
- [taskgroup-id]

**Reply by NicolasPLG** [reply]
Hola @agallardol . muchas gracias por la explicación de los pasos.

Con respecto a **[channel-id]** debería ser lo mismo que colocar $VALUE#channel y **[taskgroup-id]** debería ser lo mismo que colocar $VALUE#taskGroup :)

Salu2

**Reply by agallardol** [reply]
Correcto, pero solo si el context es un task

**Reply by NicolasPLG** [reply]
El contexto está en una task. gracias @agallardol cerraré el tema :>

**Reply by NicolasPLG** [reply]
hola @agallardol 

No funcionó la config para enviar correos a los usuarios

**1.- Copio la rutina completa**
```
{
                            "_id" : ObjectId("609000fe314fe8000b98fac5"), 
                            "key" : "send_email", 
                            "name" : "PBEmail", 
                            "data" : {
                                "subject" : "Te han asignado a una nueva solicitud", 
                                "from" : "solicitudes@municipalidadlascondes.cl", 
                                "content" : {
                                    "recipientName" : "Municipalidad Las Condes", 
                                    "recipientEmail" : "Municipalidad Las Condes", 
                                    "companyName" : "municipalidadlascondes", 
                                    "title" : [
                                        "Te han asignado la solicitud: ", 
                                        "$VALUE#name"
                                    ], 
                                    "action" : "", 
                                    "code" : "", 
                                    "messageA" : "$JOIN##(Para revisar en detalle revisa el siguiente link: https://web.cotalker.com/c/g/607f22703bce770008072d07?channel=)#($VALUE#channel)", 
                                    "messageB" : "$JOIN##(Atte. Municipalidad Las Condes)"
                                }, 
                                "targets" : "($CODE#users#_id#($CODE#task#channel#$VALUE#channel#$VALUE#taskGroup)|userList)|[map=>email]", 
                                "cc" : [
                                    "nicolas.pena@cotalker.com"
                                ], 
                                "template" : "default.html"
                            }, 
                            "next" : {
                                "DEFAULT" : ""
                            }, 
                            "customNetworkRequest" : [

                            ]
                        }```

**2.- Copio el error inicial**
```
message: "$JOIN##(Para ver mas detalles de la solicitud, ingresa al siguiente link: https://web.cotalker.com/c/g/607f22703bce770008072d07?channel=)#($VALUE#channel) ↑ ↑ ┴────┘ Unexpected end of input: ##(Para ver mas detalles de la solicitud, ingresa al siguiente link: https://web.cotalker.com/c/g/607f22703bce770008072d07?channel=)#($VALUE#channel)"```

**3.- Comentarios**: Creo que no se puede colocar una url dentro o el $VALUE#channel no es la forma correcta de escribirlo en un JOIN

**4.- Probar sin el $JOIN#**: me arroja este error
```"Error: $CODE Unknown model:task variant: channel code1: $VALUE code2: channel code3: $VALUE "
```

**5.- Comentarios:** Creo que es porque puede que haya copiado mal el código del **target**?.

**6.- Dejar sin code y sin JOIN**: Me funciona pero no me envía el correo (dejé como target el nicolás.pena@cotalker.com). Me arroja esto.

````
key: "send_email"
name: "PBEmail"
status: "DEFAULT"
result:
status: false
nextBot: ""
iteration: 5```

Alog estaría mal en mi rutina?

**Reply by NicolasPLG** [reply]
el nombre de "template" : "default.html"estaba mal escrito. al cambiarlo y probar me funcionó!:)
